src/main/java/staffconnect/model/person/Faculty.java

```diff
@@ -74,17 +77,21 @@ public Faculty(String faculty) {
     public static boolean isValidFaculty(String test) {
         requireNonNull(test, "faculty cannot be null");
 
-        return Arrays.stream(FacultyName.values())
-                .anyMatch(faculty -> faculty.getFacultyName().equalsIgnoreCase(test));
+        return Stream.of(FacultyName.values())
+                .flatMap(member -> Arrays.stream(member.facultyNameValues))
+                .anyMatch(value -> value.equalsIgnoreCase(test));
     }
 
     private static FacultyName fromString(String name) {
-        for (FacultyName faculty : FacultyName.values()) {
-            if (faculty.getFacultyName().equalsIgnoreCase(name)) {
-                return faculty;
-            }
+        Optional<FacultyName> match = Stream.of(FacultyName.values())
+                .filter(member -> Arrays.stream(member.facultyNameValues)
+                        .anyMatch(value -> value.equalsIgnoreCase(name)))
+                .findFirst();
+        if (match.isPresent()) {
+            return match.get();
+        } else {
+            throw new IllegalArgumentException("No enum constant matches the provided name: " + name);
```

Comment:
Isn't it better to simplify this to use the ```orElseThrow``` method of the optional class?

Reply:
Yeah that makes sense